R/create_default_parameters.R

```diff
@@ -0,0 +1,584 @@
+    slope_asc.value = 1,
+    slope_asc.estimated = TRUE,
+    inflection_point_desc.value = 4,
+    inflection_point_desc.estimated = TRUE,
+    slope_desc.value = 1,
+    slope_desc.estimated = TRUE
+  )
+  return(default)
+}
+
+#' Create Default Selectivity Parameters
+#'
+#' @description
+#' This function creates default fleet selectivity parameters based on the provided
+#' selectivity form. It supports different selectivity forms such as LogisticSelectivity
+#' and DoubleLogisticSelectivity.
+#' @param fleet_name A character. Name of the fleet.
+#' @param form A character. Selectivity form (e.g., LogisticSelectivity and DoubleLogisticSelectivity).
+#' @return A list containing default selectivity parameter values for the specified form.
+#' @noRd
+create_default_selectivity <- function(fleet_name, form = c("LogisticSelectivity", "DoubleLogisticSelectivity")) {
+  # Validate input fleet_name
+  if (!is.character(fleet_name) || nchar(fleet_name) == 0) {
+    cli::cli_abort("The {.var fleet_name} argument must be a non-empty character string.")
+  }
+
+  # Validate selectivity form
+  valid_forms <- c("LogisticSelectivity", "DoubleLogisticSelectivity")
+
+  default <- switch(form,
+    "LogisticSelectivity" = create_default_Logistic(),
+    "DoubleLogisticSelectivity" = create_default_DoubleLogistic(),
+    check_valid_input(input = form, valid_options = valid_forms, arg_name = "form")
+  )
+
+  names(default) <- paste0(form, ".", names(default))
+
+  return(default)
+}
+
+#' Create Default Fleet Parameters
+#'
+#' @description
+#' This function creates default parameters for a given fleet. It compiles selectivity
+#' parameters along with index and agecomp distributions.
+#'
+#' @param fleets A list of fleet configurations.
+#' @param fleet_name A character. Name of the fleet.
+#' @param data An S4 object. FIMS input data.
+#' @return A list with default parameters for the fleet.
+#' @noRd
+create_default_fleet <- function(fleets,
+                                 fleet_name,
+                                 data) {
+  # Check if the fleet_name exists in fleets list
+  check_valid_input(input = fleet_name, valid_options = names(fleets), arg_name = "fleet_name")
+
+  # Check if data is an object from FIMSFrame class
+  check_arg_data(data)
+
+  # Create default selectivity parameters
+  selectivity_form <- fleets[[fleet_name]][["selectivity"]][["form"]]
+  selectivity_default <-
+    create_default_selectivity(fleet_name, form = selectivity_form)
+
+  # Get fleet types from the data object
+  fleet_types <- data@data |>
+    subset(name == fleet_name) |>
+    (\(x) x$type)() |>
+    unique()
+
+  # Determine default Fleet parameters based on fleet type
+  process_default <- if ("landings" %in% fleet_types) {
+    list(
+      log_Fmort.value = log(rep(0.00001, data@n_years)),
+      log_Fmort.estimated = TRUE
+    )
+  } else {
+    list(
+      log_q.value = 0,
+      log_q.estimated = TRUE
+    )
+  }
```

Comment:
I do not think that being a survey or a fleet is mutually exclusive, or at least it is not supposed to be. But, will down-stream stuff fail if both F and q are present. You should be able to fit to catch data and index data from the same fleet if you want to. This has been a problem in the past with users not knowing it is possible to do so. I think I will take more time to ensure that this is true and that a list of length four here will work if both types of data are present.

Reply:
I was wrong 😞 so we will leave it as is but I have added a task to #662 to look into this.